tmt/steps/execute/internal.py

```diff
@@ -52,21 +52,38 @@ def wake(self):
         # Make sure that script is a list
         tmt.utils.listify(self.data, keys=['script'])
 
-    def _show_progress(self, test_index, number_of_tests, test_name):
+    def _show_progress(self, test_index, number_of_tests, test_name,
+                       finish=False):
+        """
+        Show an interactive progress bar in non-verbose mode.
+
+        If the output is not an interactive terminal, or progress bar is
+        disabled using an option, just output the message as info without
+        utilising \r. If finish is True, overwrite the previous progress bar.
+        """
         if self.opt('verbose'):
             # Verbose mode outputs other information, using \r to
             # create a status bar wouldn't work.
             return
-        message = f"({test_index}/{number_of_tests}) {test_name}"
+        message = f"({test_index}/{number_of_tests}) {test_name}" if not \
+            finish else ""
         if sys.stdout.isatty() and not self.opt('no-progress-bar') \
                 and not self.opt('debug'):
             # Only show progress bar in an interactive shell.
+            try:
+                # We need to completely override the previous message, add
+                # spaces if necessary.
+                message = message.ljust(len(self.previous_progress_message))
+            except AttributeError:
+                # First iteration, previous message not set
+                pass
+            self.previous_progress_message = message
```

Comment:
Could it be a private attribute? It is not supposed to be used outside of this method (which is already private), right?

Reply:
Makes sense, covered in 7619218.